tests/main/interfaces-hostname-control/task.yaml

```diff
@@ -0,0 +1,65 @@
+summary: Ensure that the hostname-control interface works
+
+details: |
+    The hostname-control interface allows to configure the system hostname
+
+prepare: |
+    echo "Install test hostname-control snap"
+    #shellcheck source=tests/lib/snaps.sh
+    . "$TESTSLIB/snaps.sh"
+    install_local test-snapd-hostname-control
+
+restore: |
+    rm -f ./*.error
+
+execute: |
+    #shellcheck source=tests/lib/systems.sh
+    . "$TESTSLIB/systems.sh"
+
+    echo "Then the plug is disconnected by default"
+    snap interfaces -i hostname-control | MATCH -- '- +test-snapd-hostname-control:hostname-control'
+    
+    echo "When the plug is connected"
+    snap connect test-snapd-hostname-control:hostname-control
+    
+    echo "And we have the hostname for the current instance"
+    hostname="$(hostname)"
```

Comment:
I think we can put this before the interface connection. AIUI it is allowed by default so it's confusing if we test it as if it was only available after we do the interface connection.

Reply:
Thanks for reviewing this @mvo5 , change already done